frontend/controller/actions/group.js

```diff
@@ -897,6 +897,70 @@ export default (sbp('sbp/selectors/register', {
       sbp('okTurtles.events/emit', OPEN_MODAL, 'AddMembers')
     }
   },
+  'gi.actions/group/removeUselessIdentityContracts': function ({ contractID, possiblyUselessContractIDs }) {
+    // NOTE: should remove the identity contracts which we don't need to sync anymore
+    //       for users who don't have any common groups, and any common DMs
+    const rootState = sbp('state/vuex/state')
+    const rootGetters = sbp('state/vuex/getters')
+    const me = rootGetters.ourIdentityContractId
+
+    const removeIdentityContracts = () => {
+      const chatRoomIDs = Object.keys(rootGetters.ourDirectMessages)
+      const groupIDs = Object.keys(rootState.contracts)
+        .filter(gID => rootState.contracts[gID].type === 'gi.contracts/group' && gID !== contractID)
+
+      if (!chatRoomIDs.length && !groupIDs.length) {
+        sbp('chelonia/contract/remove', possiblyUselessContractIDs).catch(e => {
+          console.error(`[gi.actions/group/removeUselessIdentityContracts]: ${e.name} thrown by /remove useless identity contracts:`, e)
+        })
+        return
+      }
+
+      sbp('chelonia/contract/wait', [...chatRoomIDs, ...groupIDs]).then(() => {
+        if (rootGetters.ourIdentityContractId && rootGetters.ourIdentityContractId !== me) {
+          return
+        }
+
+        const pending = Object.entries(sbp('okTurtles.eventQueue/queuedInvocations'))
+          .filter(([q]) => typeof q === 'string')
+          .flatMap(([, list]) => list)
+
+        if (pending.length) {
+          return removeIdentityContracts()
+        }
+
+        const identityContractsMapToKeepSyncing = {}
```

Comment:
If you want to re-check this, you can, but:

1. This variable needs to be set before doing this call and it needs to be in the outer scope.
2. When updating it, you can re-assign it, or you can update it directly.